### PR TITLE
Working implementation with EIP-1102 changes

### PR DESCRIPTION
**Pull request by benzap**
Reworked initialization of the browser ui to incorporate the EIP-1102 changes

- Tested against production UI while on mainnet, going to do further testing tomorrow with local testnet server+client

### CODE REVIEW

File: src/district0x/ui/events.cljs [hunk 1]
```diff
@@ -715,20 +758,24 @@
     nil))
 
 (reg-event-fx
-  :district0x.log/error
```
Comment (fbielejec) [inline src/district0x/ui/events.cljs]: use https://github.com/district0x/district-ui-logging if possible

File: src/district0x/ui/events.cljs [hunk 2]
```diff
@@ -715,20 +758,24 @@
...
-  interceptors
-  (fn [{:keys [:db]} errors]
-    (apply console :error errors)
-    nil))
-
-(reg-event-fx
```
Comment (fbielejec) [inline src/district0x/ui/events.cljs]: use https://github.com/district0x/district-ui-logging#loginfo if possible

File: src/district0x/ui/web3_fx.cljs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+
+(defn web3-legacy? []
+  (not (some-> js/window .-ethereum)))
+
+
+(reg-fx
+  ::authorize-ethereum-provider
+  (fn [{:keys [:on-accept :on-reject :on-error :on-legacy]}]
+    #_(.log js/console "Using EIP-1102 Provider? " (supports-ethereum-provider?))
```
Comment (fbielejec) [inline src/district0x/ui/web3_fx.cljs]: remove unused code

File: src/district0x/ui/events.cljs (hunk 2 above)
Comment (benzap) [reply src/district0x/ui/events.cljs]: I added the district.ui.logging library, but replacing everything is bigger refactoring job. I'll start another ticket in the meantime for future refactoring work.

**Comment by benzap** [comment]
I have no idea why the tests are failing. As a reminder to myself, i'll try and see if I can get the snapshot/revert functionality i'm using in the new ethlance exposed as a re-useable library that we could use to speed up and make tests more resilient.

**Pull request merged**